onnxruntime/core/graph/model.cc

```diff
@@ -375,6 +424,34 @@ using ::google::protobuf::io::CodedInputStream;
 using ::google::protobuf::io::FileInputStream;
 using ::google::protobuf::io::ZeroCopyInputStream;
 
+Status Model::Load(int fd, ONNX_NAMESPACE::ModelProto& model_proto) {
+  if (fd < 0) {
+    return Status(ONNXRUNTIME, INVALID_ARGUMENT, "<p_fd> less than 0.");
+  }
+
+#if GOOGLE_PROTOBUF_VERSION >= 3002000
```

Comment:
This requires extensive commentary

Reply:
Same comment as above.

Reply:
This is not a new change as will be visible by the latest commit. I removed the code duplication which implicitly implied that I added this code. We can fix them in a separate PR.